### DISCUSSION TITLE
VsCode extension?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [alternetsoft/AlternetUI#133] [Ideas]
Tying this to a windows only development IDE is not a good idea.
Can we get a extension for vscode?

**Comment by generalloki** [comment]
I remember that uixml file syntax highlighting would be good to have in the VSCode extension. 
As to project/form creation it is possible to do it using command line:
https://docs.alternet-ui.com/tutorials/hello-world/command-line/hello-world-command-line.html

@Shadowblitz16 , What features from your point of view are needed in VSCode extension for Alterent.UI?

**Comment by Shadowblitz16** [comment]
> I remember that uixml file syntax highlighting would be good to have in the VSCode extension. As to project/form creation it is possible to do it using command line: https://docs.alternet-ui.com/tutorials/hello-world/command-line/hello-world-command-line.html
> 
> @Shadowblitz16 , What features from your point of view are needed in VSCode extension for Alterent.UI?

I would need the following...
- Syntax highlighting for xaml
- Auto complete for xaml
- Display view to see how to ui would look.

**Comment by generalloki** [comment]
In order to have syntax highlight in uixml, you need:

- Create '.vscode' subfolder in you project folder
-  Create there 'settings.json' file with such content:

```json
{
	"files.associations": {
	    "*.uixml": "xml",
	}
}
```

**Comment by generalloki** [comment]
We have created simple sample which shows live uixml preview. It is cross platform. It is not embedded in Visual Studio and can work without it. Here is the screenshot and discussion:

https://github.com/alternetsoft/AlternetUI/discussions/130

For now I am moving this to discussion as there is a way to preview uixml without Visual Studio and highlight it in VsCode.

I beleive we return to this after other more important issues are fixed.
